GUI/DamageNumber/DamageNumber.gd

```diff
@@ -24,20 +29,29 @@ func set_damage_effectiveness(effectiveness: float) -> void:
 		_color = Color.orange
 	else:
 		_color = Color.red
+	_effectiveness = effectiveness
 	$Label.modulate = _color
 
 func _physics_process(delta: float) -> void:
 	_total_delta += delta
 	if _total_delta > _LIFETIME:
 		queue_free()
 	else:
-		var progress = _total_delta / _LIFETIME
-		var pos = get_global_transform().origin
-		var cam = get_tree().get_root().get_camera()
-		var screen_pos = cam.unproject_position(pos)
-		var size = _label.get_size()
-		var scale = _label.get_scale()
-		var position = screen_pos + Vector2(0, -size.y * scale.y)
-		_label.set_position(position)
-		_label.modulate.a = 1.0 - progress
-		translation += _direction * delta * _SPEED
+		_update(delta)
+
+func _update(delta: float) -> void:
```

Comment:
Can call this `project_2d` or `draw_2d`, something to do with projecting the 2d node onto the 3d space ah? Otherwise `_update` is too general liao, later we won't remember what this is for and add more stuff into it.

Reply:
okkz